### ISSUE TITLE
mvn clean doesn't clean installer directory

### ISSUE THREAD

**Issue by pameyer** [IQSS/dataverse#3974]
executing `mvn clean` doesn't remove `dvinstall.zip` in `scripts/installer` if it's present.

**Comment by pdurbin** [comment]
@pameyer as I mentioned at http://irclog.iq.harvard.edu/dataverse/2017-06-30#i_54026 I think @landreev should take a look at your pull request at (#3975) as time allows but in the meantime, can you please explain a little more either here, in [chat](http://chat.dataverse.org), or in person what you're up to? Are you worried you're going to check the zip file into the code? We could add it to the `.gitignore`. I think part of my confusion is that I never have a need to create that zip file. As far as I know, we only create at release time. I guess I could explain this a little more clearly at http://guides.dataverse.org/en/4.7/developers/making-releases.html#make-artifacts-available-for-download . You're welcome to make that zip file but why are you bothering? Why not just build the war file? Thanks.

**Comment by pameyer** [comment]
The intention here is to reduce the number of manual steps / things that can go wrong when initializing an automated deployment from the installer zip file (and starting the procedure there because it's where non-automated users would usually be starting the installation process).

It might be useful to have the makefile detect when the war file has changed (which it doesn't currently appear to - so deployment needs `mvn package; cd $installer_directory ; make`); but that's out of scope for this issue.

**Comment by pdurbin** [comment]
@pameyer right. And as you pointed out, the automated deployment I do with the phoenix server ( http://guides.dataverse.org/en/4.7/developers/testing.html#the-phoenix-server ) do **not** build the installer zip file (we need a name for that thing). Probably it should! Can we make this issue about that? If so, I'm happy to help make it happen. We'd change the issue title to something like "phoenix tests should build and use dvinstall.zip" or something.

**Comment by pdurbin** [comment]
As I just mentioned at https://github.com/IQSS/dataverse/pull/3975#issuecomment-342856718 I think what this issue highlights is that our release process is not well documented. At http://guides.dataverse.org/en/4.8.2/developers/making-releases.html we just say "upload the following artifacts to the draft release you created" but we don't explain how these artifacts (the war file and the installer zip, mostly) are created. Perhaps we should start by documenting the current process for creating these artifacts. This may more obviously highlight the problem that this issue is about.

**Comment by pdurbin** [comment]
> Perhaps we should start by documenting the current process for creating these artifacts.

I added 4a75585 to pull request #4502.

**Comment by pdurbin** [comment]
I just thought I'd mention that I'm mentoring students on #4040 and they got a little tripped up on making the "dvinstall.zip" file which is referenced in the build.sh script I use to push images to Docker Hub: https://github.com/IQSS/dataverse/blob/v4.8.5/conf/docker/build.sh#L34

**Comment by pdurbin** [comment]
@pameyer and I talked out pull request #3975. We're both a little confused by the deletion portion but figure it has to do with a recent merge from develop. The substance of the pull request is in the orginal commit at 3a6c89e but to me it would be surprising behavior (action at a distance) to have Maven clear out a directory somewhere else on the file system. Basically, we have one build tool for dealing with the Java code, which is Maven, and another tool to create the dvinstall.zip file, which is Make. I like the separation.

@pameyer has very helpfully created #4528 which expresses the problem he's facing (`make` doesn't know the war file has been updated) so let's focus our energies there. I'm closing this issue.
